renderdoc/driver/vulkan/vk_manager.cpp

```diff
@@ -682,3 +682,40 @@ bool VulkanResourceManager::ResourceTypeRelease(WrappedVkRes *res)
 {
   return m_Core->ReleaseResource(res);
 }
+
+void VulkanResourceManager::Serialise_DeviceMemoryRefs(WriteSerialiser &ser)
+{
+  SCOPED_LOCK(m_Lock);
+
+  std::vector<HalfInterval<FrameRefType>> intervals;
+
+  for(auto refs_it = m_MemFrameRefs.begin(); refs_it != m_MemFrameRefs.end(); refs_it++)
+  {
+    MemRefs &refs = refs_it->second;
+    intervals.clear();
+    intervals.reserve(refs.rangeRefs.size());
+    for(auto it = refs.rangeRefs.begin(); it != refs.rangeRefs.end(); it++)
+      intervals.push_back({it->start(), it->value()});
+
+    uint32_t chunkSize =
+        uint32_t(sizeof(ResourceId) + sizeof(std::vector<HalfInterval<FrameRefType>>::size_type) +
+                 intervals.size() * HalfInterval<FrameRefType>::packed_size);
```

Comment:
This chunk size is very tightly estimated, and I think it will break on 32-bit where the `size_type` is only 32-bit, but the array size that's serialised is always 64-bit to ease portability between captures.

It's fine for a chunk size estimate here to be an over-estimate, the only reason we need to provide it up-front in this case is because we're serialising at the end of the frame where we're streaming directly through a compressor to disk without buffering, so we can't seek that stream to fixup the length after we're done serialising.

So I'd suggest just doing a simple estimate of 32 bytes overhead + size of the inner vectors which can be based conservatively on `sizeof(HalfInterval)` even if that over-estimates its actual serialised size by a few kb.